### DISCUSSION TITLE
Does domain redirect timeout?

### DISCUSSION THREAD

**Discussion by BeigeSponge** [vercel/community#5723] [Help] [answered]
### Summary

Is there a domain redirect timeout? Currently I have a FASTAPI backend which is being called by my React frontend which is hosted on Vercel. 

I was running into CORS issues so I had a domain redirect for my backend. However, I have some long running API calls > 60 seconds which are getting 502 bad gateway errors. API calls < 60 seconds work fine (I've tested this using sleep commands)

I've checked the EC2 instance and I'm getting 499 errors, I've also checked increased the load balancer timeout to 600 seconds so the only place I can think of why the error is occurring is due to the domain redirect timing out. If there is a timeout, how can I alter it?

### Steps to Reproduce

1. Create FASTAPI call
2. Add domain redirect from api url (i.e. http://domain/api_call) to another domain (i.e. http://api.domain/api_call)
3. Use time.sleep(60) to mimic long runnning API calls.

**Comment by swarnava** [comment]
Typically you can run Serverless function upto 10 seconds on Hobby plan and 15 seconds on Pro plan. For Pro plan, you can increase Serverless function upto 300 seconds: https://vercel.com/docs/functions/configuring-functions/duration

**Reply by BeigeSponge** [reply]
@swarnava This isn't a severless function, unless I'm misunderstanding how the redirect works. The frontend hosted on vercel is calling a backend url I am hosting on a EC2 instance. Due to some CORS errors and difficulty managing the certificate, I am forwarding the requests from https://domain/api_call (hosted on Vercel) to api.domain/api_call (hosted on my EC2 instance). I think this domain redirect is timing out so when the API returns, it is no longer redirecting to the right URL. However, if this doesn't time out the error could be somewhere else in my infrastructure.

**Comment by BeigeSponge** [comment, marked as answer]
In answer to my own question, yes it does, I couldn't find it on the docs. https://vercel.com/docs/limits/overview#proxied-request-timeout

You can change this in NextJS (as discussed here: https://github.com/vercel/next.js/discussions/36598) but not when hosted with Vercel.
